hugegraph-api/src/main/java/org/apache/hugegraph/core/GraphManager.java

```diff
@@ -428,11 +439,49 @@ private void serverStarted(HugeConfig config) {
                         "The server role can't be null or empty");
         this.server = IdGenerator.of(server);
         this.role = NodeRole.valueOf(role.toUpperCase());
+
+        boolean supportRoleStateWorker = this.supportRoleStateWorker();
+        if (supportRoleStateWorker) {
+            this.role = NodeRole.WORKER;
+        }
+
         for (String graph : this.graphs()) {
             HugeGraph hugegraph = this.graph(graph);
             assert hugegraph != null;
             hugegraph.serverStarted(this.server, this.role);
         }
+
+        if (supportRoleStateWorker) {
+            this.initRoleStateWorker();
+        }
+    }
+
+    private void initRoleStateWorker() {
+        E.checkArgument(this.roleStateWorker == null, "Repetition init");
+        this.applyThread = Executors.newSingleThreadExecutor();
+        this.roleStateWorker = this.authenticator().graph().roleElectionStateMachine();
+        this.applyThread.execute(() -> {
```

Comment:
can move applyThread to RoleElectionStateMachine class?

Reply:
It's easier to write a single test with asynchronous threads outside